fdb-record-layer-lucene/src/main/java/com/apple/foundationdb/record/lucene/directory/FDBIndexInput.java

```diff
@@ -73,20 +73,49 @@ public FDBIndexInput(@Nonnull final String resourceDescription, @Nonnull final F
                 0L, 0, null);
     }
 
+    /**
+     * Constructor to create and FDBIndexInput from a file referenced in the metadata keyspace.
+     *
+     * This constructor will <b>not</b> perform an asynchronous (lookahead) to the first block and will
+     * need to be <i>sliced</i> to provide correct results.
+     *
+     * This is currently used by the LuceneOptimizedCompoundReader to not have to read its first block unless needed.
+     *
+     * @param resourceDescription opaque description of file; used for logging
+     * @param fdbDirectory FDB directory mapping
+     * @param initialOffset initialOffset
+     * @param position currentBlockPosition
+     * @throws IOException exception
+     */
+    public FDBIndexInput(@Nonnull final String resourceDescription, @Nonnull final FDBDirectory fdbDirectory, long initialOffset, long position) throws IOException {
+        super(resourceDescription);
```

Comment:
I think it would be easier to follow if this constructor would call 
`this(resourceDescription, fdbDirectory, dbDirectory.getFDBLuceneFileReferenceAsync(resourceDescription), initialOffset, position, -1, CompletableFuture.completedFuture(new byte[0]))`
And, maybe add a boolean parameter to control whether to do a seek.